lib/CFRPageActions.jsm

```diff
@@ -68,17 +90,49 @@ class PageAction {
     this._showPopupOnClick = this._showPopupOnClick.bind(this);
     this.dispatchUserAction = this.dispatchUserAction.bind(this);
 
-    this._l10n = new DOMLocalization([
-      "browser/newtab/asrouter.ftl",
-      "browser/branding/brandings.ftl",
-      "browser/branding/sync-brand.ftl",
-      "branding/brand.ftl",
-    ]);
+    this._l10n = this._createDOML10n();
 
     // Saved timeout IDs for scheduled state changes, so they can be cancelled
     this.stateTransitionTimeoutIDs = [];
   }
 
+  /**
+   * Creates a new DOMLocalization instance with the Fluent file from Remote Settings.
+   * Note that it still uses the packaged Fluent file as a fallback.
+   */
+  _createDOML10n() {
+    async function* generateBundles(resourceIds) {
+      const appLocale = Services.locale.appLocaleAsBCP47;
+      const appLocales = Services.locale.appLocalesAsBCP47;
+      const l10nFluentDir = OS.Path.join(
+        OS.Constants.Path.localProfileDir,
+        RS_DOWNLOADED_FILE_SUBDIR
+      );
+      const fs = new FileSource("cfr", [appLocale], `file://${l10nFluentDir}/`);
+      // In the case that the Fluent file has not been downloaded from Remote Settings,
+      // `fetchFile` will return `false` and fall back to the packaged Fluent file.
+      const resource = await fs.fetchFile(appLocale, "asrouter.ftl");
+      if (resource) {
+        // Use a single locale array for the bundle from Remote Settings.
+        const bundle = new FluentBundle(appLocale);
+        bundle.addResource(resource);
```

Comment:
As we talked about on the phone, we'll need to get the brands stuff into this bundle, probably like so:

```javascript
if (resource) {
  for (let bundle of L10nRegistry.generateBundles([appLocale], resourceIds.slice(1))) {
    // override, as we're the last instead of the first resource
    bundle.addResource(resource, true);
    yield bundle;
  }
}
```

Reply:
Nice catch! I've updated the patch to also include other resources into the bundle.